skuba-update/tox.ini

```diff
@@ -1,71 +1,32 @@
 [tox]
+minversion = 2.5.0
 skip_missing_interpreters = True
-skipsdist = True
-envlist =
-    check,
-    py36,
-    py34,
+envlist = check, py38, py36, py34
+skipdist = True
+ignore_basepython_conflict = True
 
 [testenv]
-whitelist_externals =
-    /bin/bash
-    /sbin/bash
-basepython =
-    check: python
-    py36: python3.6
-    py34: python3.4
-envdir =
-    check: {toxworkdir}/check
-    py36: {toxworkdir}/py36
-    py34: {toxworkdir}/py34
-passenv =
-    *
-usedevelop = True
-deps =
-    -rdev-requirements.txt
-
-[testenv:py36]
-skip_install = True
 usedevelop = True
+deps = -r{toxinidir}/dev-requirements.txt
 setenv =
-    PYTHONPATH={toxworkdir}/py36
-    PYTHONUNBUFFERED=yes
+    VIRTUAL_ENV={envdir}
+    PYTHONUNBUFFERED=1
     WITH_COVERAGE=yes
 passenv =
-    *
-deps = {[testenv]deps}
-changedir=test/unit
+    HOME
+    SKUBA_UPDATE_OPTIONS
 commands =
-    bash -c 'cd ../../ && ./setup.py develop'
-    py.test --cov=skuba_update --cov-report=term-missing \
-        --cov-fail-under=100 --cov-config .coveragerc
-
-[testenv:py34]
-skip_install = True
-usedevelop = True
-setenv =
-    PYTHONPATH={toxworkdir}/py34
-    PYTHONUNBUFFERED=yes
-    WITH_COVERAGE=yes
-passenv =
-    *
-deps = {[testenv]deps}
-changedir=test/unit
-commands =
-    bash -c 'cd ../../ && ./setup.py develop'
     py.test --cov=skuba_update --cov-report=term-missing \
         --cov-fail-under=100 --cov-config .coveragerc
 
 # Disable SC1117,SC1090 checks
 # https://github.com/koalaman/shellcheck/wiki/SC1117 (too pedantic)
 # https://github.com/koalaman/shellcheck/wiki/SC1090 (shellcheck specific limitation)
 [testenv:check]
-deps = {[testenv]deps}
-skip_install = True
-usedevelop = True
-setenv =
-    PYTHONPATH={toxworkdir}/check
-    PYTHONUNBUFFERED=yes
+basepython = python3
```

Comment:
Could probably move this to `[testenv]`?

Reply:
if we do that, tox will raise a warning (for now![0]) as other envs will try to use whatever python3 is by default and resulting in a conflict between the basepython and the environment name.

So its better to use basepython to specify an environment default python rather than making it global.

[0] https://tox.readthedocs.io/en/latest/config.html#conf-ignore_basepython_conflict

Reply:
Oh great, I now notice that my comment sent during github outage wasn't sent. But yeah, the tl;dr of what I said is in that message of itxaka.
What I wrote was something akin to:

This environment is using an external python tool, and therefore we should set the basepython to use the default host's python version. Which is why basepython is set to "python3" not a major.minor version.
For the other environments, version is inferred from the environment name.


I am now heading to my other comments I might have left during that downtime.

Reply:
Good points, thanks!